### ISSUE TITLE
[Bug?]: Vite plugin discards the original host header when proxying requests to the API port

### ISSUE THREAD

**Issue by magJ** [redwoodjs/redwood#7811]
### What's not working?

The redwood vite plugin discards the original host header when proxying requests to the API port.
This appears to be due to the [`changeOrigin` proxy configration option being set to true ](https://github.com/redwoodjs/redwood/blob/01d25afff92f1dc2a535726b632479526ab94391/packages/vite/src/index.ts#L126).

This is an issue when an app needs to see the original host.
Ie, we have an app that uses different hosts for different customers `http://customer1.app.localhost`, `http://customer2.app.localhost`, etc.

### How do we reproduce the bug?

Create a new app.
Create an API function that logs the host header.
Create multiple hosts file entries with different names for 127.0.0.1
Run the app locally, and hit your API endpoint.

See that the host header is passed through, when not using vite.
Configure the vite plugin.  
Run the app again, hit your endpoint, see that the host is now `localhost`

### What's your environment? (If it applies)

```shell
System:
    OS: macOS 13.2
    Shell: 5.8.1 - /bin/zsh
  Binaries:
    Node: 16.17.0 - /private/var/folders/ff/cqzpy9d57bb89dxb58j8wyyr0000gn/T/xfs-4cf4af7e/node
    Yarn: 3.2.1 - /private/var/folders/ff/cqzpy9d57bb89dxb58j8wyyr0000gn/T/xfs-4cf4af7e/yarn
  Databases:
    SQLite: 3.39.5 - /usr/bin/sqlite3
  Browsers:
    Chrome: 111.0.5563.64
    Edge: 109.0.1518.78
    Firefox: 110.0.1
    Safari: 16.3
  npmPackages:
    @redwoodjs/auth-dbauth-setup: 4.1.4 => 4.1.4 
    @redwoodjs/core: ^4.1.4 => 4.1.4
```

### Are you interested in working on this?

- [ ] I'm interested in working on this

**Comment by Tobbe** [comment]
Assigning to you @dac09 just because it's Vite related

**Comment by xmonkee** [comment]
+1 on this. Breaks our app which uses subdomains rendering customer-specific data. Back to using webpack for now

**Comment by dac09** [comment]
Sorry about the radio silence! Fell off my radar. Fix is in PR #9066

**Comment by magJ** [comment]
@dac09 I got a chance to try vite again as we migrate to redwood 7, I noticed that this is not fixed for production deployments, only for dev. 

`runFeServer.ts` still has `changeOrigin` set to true.
